pkg/security/probe/probe.go

```diff
@@ -309,17 +309,23 @@ func (p *Probe) Start() {
 	go p.reOrderer.Start(&p.wg)
 }
 
-// SetEventHandler set the probe event handler
-func (p *Probe) SetEventHandler(handler EventHandler) {
-	p.handler = handler
+// AddEventHandler set the probe event handler
+func (p *Probe) AddEventHandler(eventType model.EventType, handler EventHandler) {
+	p.handlers[eventType] = append(p.handlers[eventType], handler)
 }
 
 // DispatchEvent sends an event to the probe event handler
 func (p *Probe) DispatchEvent(event *Event, size uint64, CPU int, perfMap *manager.PerfMap) {
 	seclog.TraceTagf(event.GetEventType(), "Dispatching event %s", event)
 
-	if p.handler != nil {
-		p.handler.HandleEvent(event)
+	// send wildcard first
+	for _, handler := range p.handlers[model.UnknownEventType] {
+		handler.HandleEvent(event)
+	}
+
+	// send specific event
+	for _, handler := range p.handlers[event.GetEventType()] {
+		handler.HandleEvent(event)
```

Comment:
Do we need to send events to custom handlers if `AgentMonitoringEvents` is not enabled? I wonder if we could add this check here as well
```
if p.config.AgentMonitoringEvents {
}
```

Reply:
I think custom handlers are only added if enabled https://github.com/DataDog/datadog-agent/pull/11776/files#diff-3d1a943843e44b06eb76fd0d68d08665d3dafbfa6c9f5d48f928c17e2b5771c8R115 ?

Reply:
`AgentMonitoringEvents` config parameter if used by the agent to send internal health custom events

Reply:
Ohh, that makes sense. Thanks for clarifying it!!